src/genomic_features/annotate_anndata.py

```diff
@@ -0,0 +1,112 @@
+
+    Returns
+    -------
+    annotated_var
+        The annotated AnnData.var DataFrame.
+    """
+    # Pick column with IDs in annotation table
+    if id_column is None:
+        id_column = [
+            i
+            for i in annotation_df.columns
+            if annotation_df[i].is_unique and i.endswith("_id")
+        ]
+        if len(id_column) == 0:
+            raise ValueError(
+                "No unique ID column found in annotation_df - specify ID column with `id_column` parameter"
+            )
+        if len(id_column) > 1:
+            raise ValueError(
+                "Multiple unique ID columns found in annotation_df - specify ID column with `id_column` parameter"
+            )
+        else:
+            id_column = id_column[0]
+    else:
+        if not annotation_df[id_column].is_unique:
+            raise ValueError(f"Column {id_column} does not contain unique IDs")
+
+    adata_var_merge = adata_var.copy()
+
+    # Check for common columns between adata_var and annotation_df
+    # (these will be overwritten by the merge)
+    common_cols = annotation_df.columns[
+        annotation_df.columns.isin(adata_var_merge.columns)
+    ].tolist()
+    if len(common_cols) > 0:
+        warnings.warn(
+            f"Columns {common_cols} are present in both adata_var and annotation_df - these will be overwritten",
+            stacklevel=2,
+        )
+        adata_var_merge.drop(common_cols, axis=1, inplace=True)
+
+    # Pick IDs in adata_var table
+    if on is None:
+        assert (
+            adata_var_merge.index.is_unique
+        ), "var_names do not contain unique IDs - specify ID column with `on` parameter"
+        index_name = adata_var_merge.index.name
+        adata_var_merge = adata_var_merge.reset_index(names="var_names")
+    else:
+        if on not in adata_var_merge.columns:
+            raise ValueError(
+                "Column specified by `on` does not exist in adata_var_merge"
+            )
+        if not adata_var_merge[on].is_unique:
+            raise ValueError("Column specified by `on` does not contain unique IDs")
+        adata_var_merge["var_names"] = adata_var_merge[on].copy()
+
+    # Merge
+    annotated_var = (
+        adata_var_merge.reset_index()
+        .merge(annotation_df, how="left", right_on=[id_column], left_on=["var_names"])
+        .set_index("index")
+    )
+
+    # Retain order and var_names
+    if on is None:
+        annotated_var = annotated_var.set_index("var_names")
+        annotated_var = annotated_var.loc[adata_var_merge["var_names"]]
+        annotated_var.index.name = index_name
+    else:
+        annotated_var = annotated_var.loc[adata_var_merge.index]
+        annotated_var = annotated_var.drop("var_names", axis=1)
+
+    # Check for missing genes
+    missing_vars = annotated_var.index[
+        annotated_var[annotation_df.drop([id_column] + common_cols, axis=1).columns]
```

Comment:
I think this should be without the addition `common_cols`? If you only try to annotate with a data frame with two variables where one of them is a `common_cols` and the other one is the `id_column` it will say that all vars are missing (even though this is not the case) because 
`annotated_var.index[
        annotated_var[empty_list].isna().all(axis=1)]` 
returns all indices